src/main.rs

```diff
@@ -1,91 +1,186 @@
-    async fn ready(&self, ctx: Context, _data_about_bot: Ready) {
-        // Register each command name and description
-        let commands = ApplicationCommand::set_global_application_commands(&ctx.http, |commands| {
-            commands.create_application_command(|command| {
-                command.name("ping").description("A ping command")
-            })
-        })
-        .await;
-
-        println!(
-            "I now have the following global slash commands: {:#?}",
-            commands
-        );
-    }
-
-    // From docs:
-    // Dispatched when an interaction is created (e.g a slash command was used or a button was clicked).
-    // Provides the created interaction.
+    // Dispatched when an interaction is created (e.g. a slash command was used or a
+    // button was clicked). Provides the created interaction.
+    //
+    // When a new interaction is created, we check its name against all registered
+    // application commands (in [`APPLICATION_COMMANDS`]). If any match,
+    // we dispatch that application command's handler.
+    //
+    // # Panics
+    //
+    // Panics if none of the available [`APPLICATION_COMMANDS`] match, or if we run
+    // into an error handling this application command.
     async fn interaction_create(&self, ctx: Context, interaction: Interaction) {
-        // Figure out what interaction was sent
+        println!("{:#?}", interaction);
         if let Interaction::ApplicationCommand(command) = interaction {
-            let content = match command.data.name.as_str() {
-                "ping" => "Hey!".to_string(),
-                _ => "I didn't get that.".to_string(),
-            };
+            for application_command in APPLICATION_COMMANDS {
+                if application_command.get_name() == command.data.name {
+                    application_command
+                        .handle_command(&ctx, command)
+                        .await
+                        .expect("failed to handle application command");
+                    return;
+                }
+            }
 
-            // Create responses and shoot them off back to the channel
-            if let Err(why) = command
+            command
                 .create_interaction_response(&ctx.http, |response| {
-                    response
-                        .kind(InteractionResponseType::ChannelMessageWithSource)
-                        .interaction_response_data(|message| message.content(content))
+                    response.interaction_response_data(|message| {
+                        message.content(format!(
+                            "Unhandled application command \"{}\"!",
+                            command.data.name
+                        ))
+                    })
                 })
                 .await
+                .expect("failed to send message");
+        }
+    }
+
+    // Dispatched when a guild is created; or an existing guild's data is sent to
```

Comment:
Ideally we'll modify this logic to handle global application commands, but I'm using guild-specific registration for more rapid iteration, for now.

Reply:
that's fine for now - I think only @tcheinen and @ProfDoof were planning on maybe expanding this, so for our current scope that's fine